pkg/restore/client.go

```diff
@@ -408,16 +424,45 @@ func (rc *Client) GoCreateTables(
 		outCh <- rt
 		return nil
 	}
+	startWork := func(t *utils.Table, done func()) {
+		defer done()
+		if err := createOneTable(nil, t); err != nil {
+			errCh <- err
+			return
+		}
+	}
+	if len(dbPool) > 0 {
+		workers := utils.NewWorkerPool(uint(len(dbPool)), "DDL workers")
+		startWork = func(t *utils.Table, done func()) {
+			workers.ApplyWithID(func(id uint64) {
+				defer done()
+				selectedDB := int(id) % len(dbPool)
+				if err := createOneTable(dbPool[selectedDB], t); err != nil {
+					errCh <- err
+					return
+				}
+			})
+		}
+	}
+
 	go func() {
+		wg := new(sync.WaitGroup)
```

Comment:
i'd suggest we use a `errgroup` here so if one table errors, all other tables would be canceled too.

add a TODO if it's not convenient to do it here.

Reply:
Because `startWork` is asynchronous (i.e. it returns its error by sending it to `errCh`), and cannot be synchronized since tasks executing `worker` is asynchronous. If we did that, the style of our code would be like:

```go
innerErrCh = make(chan error, 1)
startWork = // some func() error

errgroup.Go(func() { 
  worker.Apply(func() { innerErrCh <- startWork() })
  return <- innerErrCh
}
```

Which (IMO) is a little dirty... In fact, when we fire issue error to `errCh`, `RunRestore` should stop execution and cancel all works in the pipeline (even sometimes with confusing error message :( ).